causalpy/pymc_experiments.py

```diff
@@ -54,12 +54,23 @@ class TimeSeriesExperiment(ExperimentalDesign):
     def __init__(
         self,
         data: pd.DataFrame,
-        treatment_time: int,
+        treatment_time: int | float | pd.Timestamp,
         formula: str,
         model=None,
         **kwargs,
     ) -> None:
         super().__init__(model=model, **kwargs)
+
+        # Input validation
+        if isinstance(data.index, pd.DatetimeIndex):
+            assert isinstance(
```

Comment:
I suggest to raise an error than using `assert`. In general is not a good practice, see https://bandit.readthedocs.io/en/latest/plugins/b101_assert_used.html#b101-test-for-use-of-assert

you could define
```
class BadIndexException(Exception):
    pass
```
and raise it when the desired condition is violated.

Reply:
Done. This is my first time using custom exceptions, but it seems to all be working ok.